src/scss/foundations/colors.scss

```diff
@@ -0,0 +1,132 @@
+@use "../lib";
+
+$_color-action-base: #067f96;
+$_color-action-dark: #e5f1f4;
+$_color-action-light: #045767;
+
+$_color-alert-base: #ffdc50;
+$_color-alert-dark: #fff8dc;
+$_color-alert-light: #ffc010;
+
+$_color-alpha-base: rgba(#21283b, 0.70);
+
+$_color-brand-base: #8334c2;
+$_color-brand-dark: #f2eaf8;
+$_color-brand-light: #5b2488;
+
+$_color-error-base: #df1654;
+$_color-error-dark: #fbe7ed;
+$_color-error-light: #b21143;
+
+$_color-neutral-25: #595d6a;
+$_color-neutral-50: #90939d;
+$_color-neutral-75: #c7c9cd;
+$_color-neutral-85: #dddee1;
+$_color-neutral-95: #f3f4f4;
+$_color-neutral-base: #21283b;
+$_color-neutral-light: #ffffff;
+
+$_color-shadow-action: rgba(#067f96, 0.40);
+$_color-shadow-base: #000000;
+$_color-shadow-brand: rgba(#8334c2, 0.40);
+$_color-shadow-dark: rgba(#595d6a, 0.40);
+
+$_color-success-base: #1d8147;
+$_color-success-dark: #e8f2ec;
+$_color-success-light: #145a31;
+
+// The color palette of Pyxis
+//
+// @type Map
+//
+// @access private
+
+$_colors: (
+  action: (
+    base: $_color-action-base,
+    dark: $_color-action-dark,
+    light: $_color-action-light,
+  ),
+  alert: (
+    base: $_color-alert-base,
+    dark: $_color-alert-dark,
+    light: $_color-alert-light,
+  ),
+  alpha: (
+    base: $_color-alpha-base,
+  ),
+  brand: (
+    base: $_color-brand-base,
+    dark: $_color-brand-dark,
+    light: $_color-brand-light,
+  ),
+  error: (
+    base: $_color-error-base,
+    dark: $_color-error-dark,
+    light: $_color-error-light,
+  ),
+  neutral: (
+    25: $_color-neutral-25,
+    50: $_color-neutral-50,
+    75: $_color-neutral-75,
+    85: $_color-neutral-85,
+    95: $_color-neutral-95,
+    base: $_color-neutral-base,
+    light: $_color-neutral-light,
+  ),
+  shadow: (
+    action: $_color-shadow-action,
+    base: $_color-shadow-base,
+    brand: $_color-shadow-brand,
+    dark: $_color-shadow-dark,
+  ),
+  success: (
+    base: $_color-success-base,
+    dark: $_color-success-dark,
+    light: $_color-success-light,
+  ),
+);
+
+// Returns the $color hex with a $variant from map $_colors
+//
+// @param {String} $color - Name of color
+// @param {String} $variant - Name of variant associated with $color, by default is "base"
+//
+// @return {Color} Representation of named color
+// @throws {Error}
+//
+// @access public
+
+@function color($color, $variant: base) {
+  @if not map-has-key($_colors, $color) {
+    @error "Unknown color `#{$color}` in map `$_colors`";
```

Comment:
Si potrebbe fare una funzione in un file `lib/logger.scss` del tipo `function keyNotFoundError($key, $map)` da riutilizzare in giro, così da avere una sorta di sistema centralizzato.

Reply:
Ho pushato un esempio di funzione più generica, ma forse preferisco avere degli errori più puntuali e precisi per ogni funzione

Reply:
Si il mio intento sarebbe solo quello di avere una funzione del tipo: 
- dammi una chiave
- dammi una mappa 
- stampo un messaggio predefinito "non trovo la chiave _chiave_ nella mappa _mappa_"

In questo modo non dobbiamo preoccuparci di avere sempre la stessa sintassi in tutte le funzioni di loggin/errore ma possiamo semplicemente invocarle. 
Pensala un po' come una funzione di logging per il nostro SCSS.